File: app/src/main/java/com/suzei/minote/ui/settings/SettingsActivity.java
package com.suzei.minote.ui.settings;

import android.content.SharedPreferences;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.PreferenceManager;

import com.suzei.minote.R;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SettingsFragment settingsFragment = SettingsFragment.newInstance();

        getSupportFragmentManager().beginTransaction()
                .replace(android.R.id.content, settingsFragment)
                .commit();

        SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(this);

        new SettingsPresenter(sharedPrefs, settingsFragment);

        setTitle("Settings");
    }
}

File: app/src/main/java/com/suzei/minote/ui/settings/SettingsActivity.kt
package com.suzei.minote.ui.settings

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.suzei.minote.Injection
import com.suzei.minote.R

class SettingsActivity : AppCompatActivity() {

    private val fragment = SettingsFragment.newInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.base_activity)
        initPresenter()
        setupFragment()
    }

    private fun initPresenter() {
        SettingsPresenter(
                fragment,
                Injection.provideSharedPreference(this)
        )
    }

    private fun setupFragment() {
        val fm = supportFragmentManager
        fm.beginTransaction().replace(R.id.base_container, fragment).commit()
    }

}